velox/functions/prestosql/aggregates/tests/ApproxDistinctTest.cpp

```diff
@@ -407,5 +407,36 @@ TEST_F(ApproxDistinctTest, mergeWithEmpty) {
   ASSERT_EQ(readSingleValue(op).value<TypeKind::BIGINT>(), 499);
 }
 
+TEST_F(ApproxDistinctTest, toIntermediate) {
+  constexpr int kSize = 1000;
+  auto input = makeRowVector({
+      makeFlatVector<int32_t>(kSize, folly::identity),
+      BaseVector::createConstant(BIGINT(), 1ll, kSize, pool()),
+  });
+  auto plan = PlanBuilder()
+                  .values({input})
+                  .singleAggregation({"c0"}, {"approx_set(c1)"})
+                  .planNode();
+  auto digests = AssertQueryBuilder(plan).copyResults(pool());
+  plan =
+      PlanBuilder()
+          .values({
+              std::static_pointer_cast<RowVector>(digests->slice(0, kSize / 2)),
+              std::static_pointer_cast<RowVector>(
+                  digests->slice(kSize / 2, kSize / 2)),
+          })
+          .partialAggregation({"c0"}, {"merge(a0)"})
+          .finalAggregation()
+          .project({"c0", "cardinality(a0)"})
+          .planNode();
+  AssertQueryBuilder queryBuilder(plan);
+  queryBuilder.config(core::QueryConfig::kAbandonPartialAggregationMinRows, 1)
+      .config(core::QueryConfig::kAbandonPartialAggregationMinPct, 0)
+      .config(core::QueryConfig::kMaxPartialAggregationMemory, 0)
+      .config(core::QueryConfig::kMaxExtendedPartialAggregationMemory, 0)
+      .maxDrivers(1);
+  queryBuilder.assertResults(input);
```

Comment:
I guess there is no way to actually verify that custom toIntermediate was called.

Reply:
Unless we put a `TestValue` there

Reply:
Actually, testAggregations helper method already includes this kind of testing. Just pass a multi-batch input.